code/modules/tutorials/flock_tutorial.dm

```diff
@@ -0,0 +1,421 @@
+
+	PerformAction(action, context)
+		if (action == FLOCK_ACTION_DRONE_RELEASE)
+			finished = TRUE
+			return TRUE
+
+/datum/tutorialStep/flock/kill
+	name = "Eliminate threat"
+	instructions = "That human has just violated causality to teleport right into your flock! Mark them for elimination using your \"designate enemy\" ability and watch as your drones attack."
+
+	SetUp()
+		..()
+		var/turf/T = locate(src.ftutorial.center.x, src.ftutorial.center.y + 3, src.ftutorial.center.z)
+		src.ftutorial.portal_in(T, /mob/living/carbon/human/normal/assistant)
+
+	PerformAction(action, context)
+		if (action == FLOCK_ACTION_MARK_ENEMY || action == FLOCK_ACTION_START_CONVERSION)
+			return TRUE
+		if (action == FLOCK_ACTION_CAGE)
+			finished = TRUE
+			return TRUE
+
+/datum/tutorialStep/flock/place_sentinel
+	name = "Construct Sentinel"
+	instructions = "There may be more humans around, build a Sentinel for protection. Move over the marked turf and use your \"place tealprint\" ability to place one."
+	var/turf/location = null
+
+	SetUp()
+		..()
+		src.location = locate(src.ftutorial.center.x, src.ftutorial.center.y - 3, src.ftutorial.center.z)
+		src.location.UpdateOverlays(marker, "marker")
+
+	PerformAction(action, context)
+		if (action == FLOCK_ACTION_TEALPRINT_PLACE && context == /obj/flock_structure/sentinel)
+			if (get_turf(src.ftutorial.fowner) == src.location)
+				src.finished = TRUE
+				return TRUE
+			else
+				return "<span class='alert'><b>You must place the tealprint on the marked tile.</b></span>"
+
+/datum/tutorialStep/flock/deposit_sentinel
+	name = "Direct drones to construct"
+	instructions = "Nearby drones will slowly automatically deposit their resources into tealprints, but you can override their torpid cognition to order them directly. Click on a drone to select it, then click on the tealprint to target it. The same works to order a drone to convert, attack, cage and so on depending what you click on."
+
+	PerformAction(action, context)
+		if (action == FLOCK_ACTION_DRONE_SELECT)
+			return TRUE
+		if (action == FLOCK_ACTION_DRONE_ORDER && context == /datum/aiTask/sequence/goalbased/flock/deposit/targetable)
+			src.finished = TRUE
+			return TRUE
+
+/datum/tutorialStep/flock/build_thing
+	var/turf/location = null
+	var/structure_type = null
+
+	PerformAction(action, context)
+		if (action in list(FLOCK_ACTION_START_CONVERSION, FLOCK_ACTION_DRONE_SELECT, FLOCK_ACTION_DRONE_ORDER))
+			return TRUE
+		if (action == FLOCK_ACTION_TEALPRINT_PLACE && context == src.structure_type)
+			if (get_turf(src.ftutorial.fowner) == src.location)
+				return TRUE
+			else
+				return "<span class='alert'><b>You must place the tealprint on the marked tile.</b></span>"
+		if (action == FLOCK_ACTION_TEALPRINT_COMPLETE)
+			var/obj/flock_structure/struct = context
+			struct.process(200) //force a high mult process to immediately charge the structure if it needs it
```

Comment:
Also seems like this is only used for Sentinels at the moment, maybe just allow the Sentinel to charge for that, or could access charge on Sentinel?

Reply:
It's used for interceptors too (actually it has to be called twice for them because of the way their process works 💀), and I'm wary of shoving infinity into things like this in case it breaks something.